repos/system_upgrade/el7toel8/actors/enableddeviceciofreeservices390/tests/test_enableddeviceciofreeservice.py

```diff
@@ -0,0 +1,37 @@
+import pytest
+
+from leapp.libraries.actor import enabledeviceciofreeservice
+from leapp.libraries.common.config import architecture
+from leapp.libraries.common.testutils import CurrentActorMocked, produce_mocked
+from leapp.libraries.stdlib import api
+from leapp.models import SystemdServicesTasks
+
+
+def test_task_produced_on_s390(monkeypatch):
+
+    def matches_architecture_mocked(arch):
+        return True
+
+    monkeypatch.setattr(architecture, 'matches_architecture', matches_architecture_mocked)
```

Comment:
@matejmatuska mocking the architecture in this way is not ideal. it's better to specify the architecture you want to have when the test is running via the CurrentActorMocked. It's more easier and you can then simply check the behaviour even for all architectures

Reply:
@matejmatuska the code is updated. you can take a look what I meant :)